.github/workflows/component-test-report.yml

```diff
@@ -0,0 +1,68 @@
+name: Component Test Reporter
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+permissions:
+  id-token: write # allows the JWT to be requested from GitHub's OIDC provider
+  contents: read # This is required for actions/checkout
+
+jobs:
+  test_and_upload:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_DEV_ACCOUNT_ID }}:role/${{ secrets.AWS_DEV_S3_SYNC_ROLE }}
+          aws-region: us-east-1
+
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          fetch-depth: 1
+
+      - name: Setup Node
+        uses: actions/setup-node@v4.0.1
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Run Tests and Generate Report
+        run: |
+          npm run test:ts -- component
+
+
+      - name: Uplaod Report to S3
+        run: |
+          aws s3 cp ./test_reports/ s3://test-integrations-dev/integrations-test-reports/${{ github.event.number }}/ --recursive
```

Comment:
should we add GitHub repo name also in the path:
s3://test-integrations-dev/integrations-test-reports/`rudder-transformer`/${{ github.event.number }}/

Please adjust the URL in the next step also after this

Reply:
Addressed